### ISSUE TITLE
MirageOS support?

### ISSUE THREAD

**Issue by tg-x** [tomjridge/tjr_kv#1]
Is there support for MirageOS?
Seen it mentioned in the slides and wondering if it's implemented or planned.

What's the status of the project otherwise in terms of feature completeness and stability?

**Comment by tomjridge** [comment]
The intention is that the code should be usable by Mirage. However, there hasn't been any interest in using it with Mirage, so I haven't put any effort into making it ACTUALLY usable.

At the moment, there is no garbage collection (so heavy modification of the store will eat up disk blocks without releasing any). Keys and values also have to be "small" (ie at least we must be able to fit a key and a value in a single block), so this precludes storing arbitrary data. There is also no guaranteed crash resilience (this would need a bit of thinking about).

Basically, as soon as people ask me for features I can start to work on them. Otherwise I am trying to implement the overall "ImpFS" filesystem (which is the real goal).

Thank you for your interest, and please let me know if there is anything you want done.
